### PR TITLE
Aden 3696 intrestitial mercury - Proof of concept

### PR DESCRIPTION
**Pull request by bognix**
## Links

* http://wikia-inc.atlassian.net/browse/ADEN-3696
* https://github.com/Wikia/app/pull/10998

## Description

Make INVISIBLE_HIGH_IMPACT_2 work on Mercury

### CODE REVIEW

File: front/main/app/styles/module/_ads.scss [hunk 1]
```diff
@@ -82,3 +82,62 @@
...
+	position: fixed;
+	top: 0;
+	width: 100%;
+}
+
+#InvisibleHighImpactWrapper {
+	z-index: $z-modal-dialog;
+
+	#INVISIBLE_HIGH_IMPACT_2 {
+		display: block !important;
```
Comment (bognix) [inline front/main/app/styles/module/_ads.scss]: This one is to overwrite this rule https://github.com/Wikia/mercury/blob/308b895ca8f921715c87f1aeae8dac50af3de66b/front/main/app/styles/module/_ads.scss#L11. I don't like it, can we do https://github.com/Wikia/mercury/blob/308b895ca8f921715c87f1aeae8dac50af3de66b/front/main/app/styles/module/_ads.scss#L11 rule less general?

File: front/main/app/templates/components/ads/invisible-high-impact-2.hbs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<div id="InvisibleHighImpactWrapper" class="hidden">
+	<div class="background"></div>
+	<a class="close">
+		{{svg 'close' role='img' class='icon close-button'}}
+	</a>
+	<div class="label">Advertisement</div>
```
Comment (bognix) [inline front/main/app/templates/components/ads/invisible-high-impact-2.hbs]: I believe this message should be localized. Not sure if it should be done at this point

File: front/main/app/components/ad-slot.js [hunk 3]
```diff
@@ -54,7 +54,10 @@ export default Component.extend(
 
 			if (this.get('isAboveTheFold')) {
 				Logger.info('Injected ad', name);
-				ads.addSlot(name);
+				ads.addSlot({
+					slotName: name,
+					onSuccess: this.get('onSuccess')
```
Comment (bognix) [inline front/main/app/components/ad-slot.js]: this is my idea of reproducing the same functionality as in https://github.com/Wikia/app/blob/dev/extensions/wikia/AdEngine/js/slot/highImpact.js#L16

I'm open to other ideas.

File: front/main/app/styles/module/_ads.scss (hunk 1 above)
Comment (jacekjursza) [inline front/main/app/styles/module/_ads.scss]: yes, maybe we should. 👍 
or use
display: block !importaner

File: front/main/app/templates/components/ads/invisible-high-impact-2.hbs (hunk 2 above)
Comment (jacekjursza) [inline front/main/app/templates/components/ads/invisible-high-impact-2.hbs]: True. Let's create ticket for that.

File: front/main/app/components/ads/invisible-high-impact-2.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+	/**
+	 * This method is called by callAllCallbacks method defined in mercury_ads_js package from MW.
+	 * This method's context is object created in adSlot.js in MW (from mercury_ads_js),
+	 * hence `this` is different than expected
+	 */
+	onSuccess() {
+		const iframe = document.getElementById(this.name).querySelector('div:not(.hidden) > div[id*="_container_"] iframe');
+
+		if (iframe.contentWindow.document.readyState === 'complete') {
```
Comment (jacekjursza) [inline front/main/app/components/ads/invisible-high-impact-2.js]: what do you think about caching iframe.contentWindow.document in some local variable? the rest of the code would became shorter and more readable

**Comment by bognix** [comment]
Closing this PR, all comments are going to be addressed in https://github.com/Wikia/mercury/pull/2695

**Pull request closed without merging**